src/platform/packages/shared/kbn-workflows/spec/schema.ts

```diff
@@ -56,10 +56,16 @@ export function getOnFailureStepSchema(stepSchema: z.ZodType, loose: boolean = f
   return schema;
 }
 
+export const CollisionStrategySchema = z.enum(['queue', 'drop', 'cancel-in-progress']);
+export type CollisionStrategy = z.infer<typeof CollisionStrategySchema>;
+
 export const WorkflowSettingsSchema = z.object({
   'on-failure': WorkflowOnFailureSchema.optional(),
   timezone: z.string().optional(), // Should follow IANA TZ format
   timeout: DurationSchema.optional(), // e.g., '5s', '1m', '2h'
+  concurrency_key: z.string().optional(), // e.g., '{{ event.host.name }}'
```

Comment:
NIT: we previously used kebab-case for yaml properties (e.g. `connector-id`, `on-failure`, `max-attempts`). idk if we want to continue to stick with kebab-case or migrate to snake-case? cc @shahargl @talboren @tinnytintin10